src/sonic-yang-models/tests/yang_model_tests/test_yang_model.py

```diff
@@ -48,272 +48,16 @@ def initTest(self):
-                'eStr': self.defaultYANGFailure['None']
-            },
-            'SNAT_WITH_CORRECT_USED_VALUE': {
-                'desc': 'SNAT_WITH_CORRECT_USED_VALUE no failure.',
-                'eStr': self.defaultYANGFailure['None']
-            },
-            'SNAT_WITH_WRONG_THRESHOLD_TYPE': {
-                'desc': 'SNAT_WITH_WRONG_THRESHOLD_TYPE pattern failure.',
-                'eStr': self.defaultYANGFailure['Pattern'] + ['wrong']
-            },
-            'DNAT_WITH_WRONG_PERCENTAGE': {
-                'desc': 'DNAT_WITH_WRONG_PERCENTAGE must condition failure.',
-                'eStr': self.defaultYANGFailure['Must']
-            },
-            'DNAT_WITH_HIGH_THRESHOLD_ERR': {
-                'desc': 'DNAT_WITH_HIGH_THRESHOLD_ERR must condition failure \
-                    about high threshold being lower than low threshold.',
-                'eStr': ['high_threshold should be more than low_threshold']
-            },
-            'DNAT_WITH_CORRECT_FREE_VALUE': {
-                'desc': 'DNAT_WITH_CORRECT_FREE_VALUE no failure.',
-                'eStr': self.defaultYANGFailure['None']
-            },
-            'DNAT_WITH_CORRECT_USED_VALUE': {
-                'desc': 'DNAT_WITH_CORRECT_USED_VALUE no failure.',
-                'eStr': self.defaultYANGFailure['None']
-            },
-            'DNAT_WITH_WRONG_THRESHOLD_TYPE': {
-                'desc': 'DNAT_WITH_WRONG_THRESHOLD_TYPE pattern failure.',
-                'eStr': self.defaultYANGFailure['Pattern'] + ['wrong']
-            },
-            'IPMC_WITH_WRONG_PERCENTAGE': {
-                'desc': 'IPMC_WITH_WRONG_PERCENTAGE must condition failure.',
-                'eStr': self.defaultYANGFailure['Must']
-            },
-            'IPMC_WITH_HIGH_THRESHOLD_ERR': {
-                'desc': 'IPMC_WITH_HIGH_THRESHOLD_ERR must condition failure \
-                    about high threshold being lower than low threshold.',
-                'eStr': ['high_threshold should be more than low_threshold']
-            },
-            'IPMC_WITH_CORRECT_FREE_VALUE': {
-                'desc': 'IPMC_WITH_CORRECT_FREE_VALUE no failure.',
-                'eStr': self.defaultYANGFailure['None']
-            },
-            'IPMC_WITH_CORRECT_USED_VALUE': {
-                'desc': 'IPMC_WITH_CORRECT_USED_VALUE no failure.',
-                'eStr': self.defaultYANGFailure['None']
-            },
-            'IPMC_WITH_WRONG_THRESHOLD_TYPE': {
-                'desc': 'IPMC_WITH_WRONG_THRESHOLD_TYPE pattern failure.',
-                'eStr': self.defaultYANGFailure['Pattern'] + ['wrong']
-            }
-        }
+        self.ExceptionTests = { }
+
+        for test_file in glob("./tests/yang_model_tests/tests/*.json"):
+            try:
+                with open(test_file) as fp:
+                    log.info("Loading file " + test_file)
+                    self.ExceptionTests.update(json.load(fp))
```

Comment:
This json load should be fine, since test json files should not grow super large.  But lets see, if we can find better way, so we do not load entire test config files in a single shot. I think we can have a section for test_config_file in tests/*.json, and while parsing here, we can create self.testFile = {} as well.